File: apollo/src/main/java/com/lsxiao.apllo/entity/Event.java
package com.lsxiao.apllo.entity;

/**
 * author lsxiao
 * date 2016-08-07 20:00
 */
public class Event {
    private final String tag;
    private final Object data;
    private final Boolean sticky;

    public Event(String tag, Object data, boolean sticky) {
        this.tag = tag;
        this.data = data;
        this.sticky = sticky;
    }

    public Event(String tag, Object data) {
        this(tag, data, false);
    }

    public Object getData() {
        return data;
    }

    public String getTag() {
        return tag;
    }

    public Boolean isSticky() {
        return sticky;
    }
}

File: core/src/main/java/com/apollo/core/entity/Event.kt
package com.apollo.core.entity

import java.io.Serializable
import kotlin.properties.Delegates

/**
 * author lsxiao
 * date 2016-08-07 20:00
 */
class Event constructor() : Serializable {
    constructor(tag: String, data: Any, pid: Int, isSticky: Boolean = false) : this() {
        this.tag = tag
        this.data = data
        this.pid = pid
        this.isSticky = isSticky
    }

    var tag: String by Delegates.notNull()
    var data: Any by Delegates.notNull()
    var pid: Int by Delegates.notNull()
    var isSticky: Boolean by Delegates.notNull()
}
